src/lib/list/selection-list.spec.ts

```diff
@@ -0,0 +1,353 @@
+import {By} from '@angular/platform-browser';
+import {MdSelectionList, MdListOption, MdListModule} from './index';
+import {createKeyboardEvent} from '@angular/cdk/testing';
+import {UP_ARROW, DOWN_ARROW, SPACE} from '../core/keyboard/keycodes';
+import {Platform} from '../core/platform/index';
+
+describe('test normal selection list with list option', () => {
+  let fixture: ComponentFixture<SelectionListWithListOptions>;
+  let listOption: DebugElement[];
+  let listItemEl: DebugElement;
+  let selectionList: DebugElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MdListModule],
+      declarations: [
+        SelectionListWithListOptions,
+        SelectionListWithCheckboxPositionAfter,
+        SelectionListWithListDisabled,
+        SelectionListWithOnlyOneOption
+      ],
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(SelectionListWithListOptions);
+    listOption = fixture.debugElement.queryAll(By.directive(MdListOption));
+    listItemEl = fixture.debugElement.query(By.css('.mat-list-item'));
+    selectionList = fixture.debugElement.query(By.directive(MdSelectionList));
+    fixture.detectChanges();
+  }));
+
+  it('should add and remove focus class on focus/blur', () => {
+    expect(listItemEl.nativeElement.classList).not.toContain('mat-list-item-focus');
+
+    listOption[0].componentInstance._handleFocus();
+    fixture.detectChanges();
+    expect(listItemEl.nativeElement.className).toContain('mat-list-item-focus');
+
+    listOption[0].componentInstance._handleBlur();
+    fixture.detectChanges();
+    expect(listItemEl.nativeElement.className).not.toContain('mat-list-item-focus');
+  });
+
+  it('should be able to dispatch one selected item', () => {
+    let testListItem = listOption[2].injector.get<MdListOption>(MdListOption);
+    let selectList = selectionList.injector.get<MdSelectionList>(MdSelectionList).selectedOptions;
+
+    expect(selectList.selected.length).toBe(0);
+
+    testListItem.toggle();
+    fixture.detectChanges();
+
+    expect(listOption[2].nativeElement.getAttribute('aria-selected')).toBe('true');
+    expect(selectList.selected.length).toBe(1);
+  });
+
+  it('should be able to dispatch multiple selected items', () => {
+    let testListItem = listOption[2].injector.get<MdListOption>(MdListOption);
+    let testListItem2 = listOption[1].injector.get<MdListOption>(MdListOption);
+    let selectList = selectionList.injector.get<MdSelectionList>(MdSelectionList).selectedOptions;
+
+    expect(selectList.selected.length).toBe(0);
+
+    testListItem.toggle();
+    fixture.detectChanges();
+
+    testListItem2.toggle();
+    fixture.detectChanges();
+
+    expect(selectList.selected.length).toBe(2);
+    expect(listOption[2].nativeElement.getAttribute('aria-selected')).toBe('true');
+    expect(listOption[1].nativeElement.getAttribute('aria-selected')).toBe('true');
```

Comment:
There should also be assertions somewhere where you check `aria-selected` being false (right now you only ever check that it's true)

(same for aria-disabled)